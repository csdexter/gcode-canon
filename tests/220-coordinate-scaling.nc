(testing coordinate systems)
(machine setup)
M05 S600
M09
M23
M49
M69

(turn servos ON)
M17

(control setup)
G15
G17
G23
G40
G49
G50
G69

(in mm)
G21

(absolute coordinates)
G90

(scaling)
G01 X10 Y10 Z10 F600
X15
Y12 X8
X10 Y10
(scale by 2 around 11,11 and do it again)
G51 X11 Y11 Z10 I2 J2 K1
X15
Y12 X8
X10 Y10
(scale by 0.5 around 11,11 and do it again)
G51 X11 Y11 Z10 I0.5 J0.5 K1
X15
Y12 X8
X10 Y10
(disable scaling)
G50

M02