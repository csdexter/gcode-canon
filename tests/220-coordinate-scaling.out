MSG: WAR: Machine servos activated!
MSG: STA: Scanning input for programs (O words)
MSG: WAR: Machine servos activated!
MPOS,10.00,10.00,10.00
MPOS,15.00,10.00,10.00
MPOS,8.00,12.00,10.00
MPOS,10.00,10.00,10.00
MPOS,19.00,9.00,10.00
MPOS,5.00,13.00,10.00
MPOS,9.00,9.00,10.00
MPOS,13.00,10.50,10.00
MPOS,9.50,11.50,10.00
MPOS,10.50,10.50,10.00
